subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/projectmodule/DefaultProjectDependencyPublicationResolver.java

```diff
@@ -79,17 +79,19 @@ public <T> T resolve(Class<T> coordsType, ProjectInternal project) {
         // Select all entry points. An entry point is a publication that does not contain a component whose parent is also published
         Set<SoftwareComponent> ignored = new HashSet<>();
         for (ProjectComponentPublication publication : publications) {
-            if (publication.getComponent() != null && publication.getComponent() instanceof ComponentWithVariants) {
-                ComponentWithVariants parent = (ComponentWithVariants) publication.getComponent();
+            SoftwareComponent component = publication.getComponent().getOrNull();
+            if (component instanceof ComponentWithVariants) {
+                ComponentWithVariants parent = (ComponentWithVariants) component;
```

Comment:
💭 Having variable named `parent` set equal the component itself is confusing, there is no parent relationship here.  Maybe just move the cast inline when adding to `ignored`.

Reply:
This is a parent-child relationship here, it is even noted in the existing comment a few lines up. The naming of `ComponentWithVariants` is confusing, but the variants from `getVariants` are not traditional "variants" as we call them usually. They are child `SoftwareComponents`.

Reply:
No, I get that, I understand this is fine, I'm saying line 84 specifically caused me to do a double-take.  

Usually you'd expect to see `var parent = component.getParent()`, not `var parent = (cast) component`.  It really _looks_ incorrect at first, like a call was forgotten, because of the names.

Reply:
I've renamed the existing `ignored` list to `children` to make this more clear